Automattic/jetpack
# Boost: use the correct plugin directory for the path in advanced-cache.php

The Cache module in Boost creates a file called wp-content/advanced-cache.php to load the caching engine before WordPress loads. During development, the plugin may use different plugin directory names so this PR changes the path to suit that.
This PR also checks that Boost_File_Cache.php exists before loading it, to avoid PHP errors if a site owner deleted the plugin without deactivating/uninstalling it.

The advanced-cache.php could define a constant if an error was found. Later code could check for that constant and report the problem with caching to the site owner.

Fixes #35288

## Proposed changes:
* Get the plugin directory using plugin_dir_path and dirname.
* Replace the hardcoded "boost" string with $plugin_name.


### Other information:


## Jetpack product discussion
pc9hqz-2rA-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Apply this PR
* Test it in an environment that doesn't use the standard jetpack-boost directory.
* Delete wp-content/advaned-cache.php if it exists.
* Visit the Boost settings page
* Check wp-content/advaned-cache.php has the correct path to load Boost_File_Cache.php
